src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -69,7 +71,7 @@ internal partial class TagComputer
             // get called.
 
             private readonly object _gate = new();
-            private (ITextSnapshot? lastSnapshot, Document? lastDocument, SyntaxNode? lastRoot) _lastProcessedData;
+            private (ITextSnapshot lastSnapshot, SumType<SyntaxNode, Document> lastDocumentOrRoot)? _lastProcessedData;
```

Comment:
i would intentionally doc that we want to drop teh Doc in the case where we have the root to ensure we're not holding onto old solutions.  Then maybe a note that this could potentially be bad for other languages, but we can revisit as needed.